ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/UseNamedAccessAction.fs

```
@@ -0,0 +1,107 @@
+    inherit FSharpContextActionBase(dataProvider)
+    
+    let mutable names = Array.empty
+    let mutable tuplePat : ITuplePat = null
+
+    override x.Text = "Use named access"
+
+    override x.IsAvailable _ =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then false else
+        // We expect a single IParenPat here
+        if pattern.Parameters.Count <> 1 then false else
+        let parenPat =
+            match pattern.Parameters.[0] with
+            | :? IParenPat as parenPat -> parenPat
+            | _ -> null
+
+        if isNull parenPat then false else
+        // This should only work when there are multiple items
+        tuplePat <- parenPat.Pattern :?> ITuplePat
+        if isNull tuplePat then false else
+        
+        let hasUnderscore =
+            tuplePat.Patterns
+            |> Seq.exists (fun p ->
+                match p with
+                | :? IWildPat -> true
+                | _ -> false)
+    
+        // We need something that is currently ignored
+        if not hasUnderscore then false else
+
+        // We need to be sure that the pattern is a DU with all named fields
+        let fcsUnionCase = pattern.ReferenceName.Reference.GetFcsSymbol().As<FSharpUnionCase>()
+        if isNull fcsUnionCase then false else
+            
+        names <-
+            fcsUnionCase.Fields
+            |> Seq.choose (fun field -> if field.IsNameGenerated then None else Some field.Name)
+            |> Seq.toArray
+        
+        // All fields need to be named
+        if names.Length <> fcsUnionCase.Fields.Count then false else
+        // The amount of fields needs to match with the tuple length
+        if names.Length <> tuplePat.Patterns.Count then false else
+
+        true
+
+    override x.ExecutePsiTransaction(_,_) =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then null else
```

Please throw an exception instead. This code should only ever run if `IsAvailable` is `true`, and that would mean the pattern should be accessible.
```suggestion
        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>().NotNull()
```